nitro-protocol/gas-benchmarks/jestSetup.ts

```diff
@@ -21,6 +21,28 @@ declare global {
 const logFile = './hardhat-network-output.log';
 const hardHatNetworkEndpoint = 'http://localhost:9546'; // the port should be unique
 
+async function waitForDuration(durationMS: number): Promise<void> {
+  await new Promise(resolve => setTimeout(resolve, durationMS));
+}
+
+async function waitUntilNodeReady(url: string): Promise<void> {
+  let statusCode = 0;
+
+  while (statusCode !== 200) {
+    try {
+      statusCode = (await axios.get(url)).status;
+    } catch (e) {
+      // It seems the hardhat node will refuse connections even when it is ready
+      // We take the refusal to mean that the node is ready otherwise we could be waiting for a long time
+      if (axios.isAxiosError(e) && e.code === 'ECONNREFUSED') {
+        return;
```

Comment:
That is a very confusing behavior...

Reply:
Agreed! I think there's something more going on here but for now this appears to work 🤷